nnvm/python/nnvm/frontend/tensorflow.py

```diff
@@ -974,24 +985,28 @@ def from_tensorflow(self, graph):
                 # Pass the node name too in attr
                 attr["_node_name"] = node.name
 
-                #ToDo: Some of the tensorflow operators internaly maintain
-                #execution layers and its output name will the layer number along with
-                #graph node name.eg: Node name:- 'Model/RNN/cell_0/RnnCell', but the
-                #output name will be 'Model/RNN/cell_0/RnnCell:0'. In this case,
-                #the digit has to be ignored.
-                if ":" in node.input[0]:
-                    in_name, _ = node.input[0].split(':')
-                    node.input[0] = in_name
-
-                # Fill shapes for all inputs in a list
-                try:
-                    inputs = [self._nodes[i] for i in node.input]
-                    for i in node.input:
-                        input_shapes[self._nodes[i]] = self._output_shapes[i]
-                    attr['_input_shapes'] = input_shapes
-                except KeyError:
-                    # TODO: Need to find clean way to handle '^CheckNumerics'
-                    pass
+                inputs = []
+                for node_input_name in node.input:
+                    node_input_key = node_input_name.split(':')
+                    slot_num = 0
+                    if len(node_input_key) > 1:
+                        slot_num = int(node_input_key[1])
+                    node_input_key = node_input_key[0]
+
+                    try:
+                        try:
+                            input_sym = self._nodes[node_input_key].__getitem__(slot_num)
+                        except NNVMError:
+                            # TODO: Fancy node name invalid slot neglect
+                            input_sym = self._nodes[node_input_key].__getitem__(0)
```

Comment:
Can you share details on this fall back?
I see slot_num is initialized to 0 above already.

Reply:
Some of Layer implementation in frontend(LSTM, Ptb) this assign fancy node name (eg. Model/RNN/cell_0/RnnCell:6 ), in this case 6 is slot updated as per our logic and only one node will be present, in this fancy naming convention we need to parse the default(0) node.